src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -16,18 +31,23 @@ export const TodoFilter = () => (
         type="text"
         className="input"
         placeholder="Search..."
+        value={query}
+        onChange={event => onQuery(event.target.value)}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
       <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
+        {query && (
+          // eslint-disable-next-line jsx-a11y/control-has-associated-label
```

Comment:
Try to fix eslint rule and remove comment

Reply:
done